GeosCore/tomas_mod.F90

```diff
@@ -4051,130 +3965,160 @@ SUBROUTINE SOACOND( MSOA, I, J, L, BOXVOL, TEMPTMS, PRES, &
+       DO K = 1, IBINS
+          MPO = 0.e+0_fp
+          DO JC = 1, ICOMP-IDIAG
+             MPO = MPO + MK(K,JC)  ! Accumulate dry mass
+          ENDDO
+          MABS(K) = MSOA * avgfrac(K)
 
-       IF ( Nk(K) > 0.e+0_fp ) THEN
-          MPO = MPO / Nk(K)
-          OCTAU(K) = 1.5e+0_fp * ( ( ( MPO + MABS(K)/Nk(K) ) ** TDT ) - &
-                                   (   MPO                   ** TDT )   )
+          IF ( Nk(K) > 0.e+0_fp ) THEN
+             MPO = MPO / Nk(K)
+             OCTAU(K) = 1.5e+0_fp * ( ( ( MPO + MABS(K)/Nk(K) ) ** TDT ) - &
+                                      (   MPO                   ** TDT )   )
 
-          ! Error checking for negative Tau
-          IF ( OCTAU(K) < 0.e+0_fp ) THEN
-             IF ( ABS(OCTAU(K)) < 1.e+0_fp ) THEN
-                OCTAU(K)=1.e-50_fp  !0.e+0_fp  !try change to tiny number instead of 0e+0_fp (win, 5/28/06)
-             ELSE
-                PRINT *,' ######### Subroutine SOACOND:  NEGATIVE TAU'
-                PRINT *,'Error at',i,j,l,'bin',k
-                PRINT *,'octau(k)',octau(k)
-                CALL ERROR_STOP( 'Negative Tau','SOACOND:2' )
+             ! Error checking for negative Tau
+             IF ( OCTAU(K) < 0.e+0_fp ) THEN
+                IF ( ABS(OCTAU(K)) < 1.e+0_fp ) THEN
+                   ! change to tiny number instead of 0e+0_fp (win, 5/28/06)
+                   OCTAU(K)=1.e-50_fp
+                ELSE
+                   PRINT *,' ######### Subroutine SOACOND:  NEGATIVE TAU'
+                   PRINT *,'Error at',i,j,l,'bin',k
+                   PRINT *,'octau(k)',octau(k)
+                   CALL ERROR_STOP( 'Negative Tau','SOACOND:2' )
+                ENDIF
              ENDIF
+
+          ELSE
+             OCTAU(K) = 0.e+0_fp
           ENDIF
+       ENDDO
+
+       ! Call condensation algorithm
+       ! Swap into Nko, Mko
+       Mko(:,:) = 0.e+0_fp
+       DO K = 1, IBINS
+          Nko(K) = Nk(K)
+          DO JC = 1, ICOMP-IDIAG    ! Now do SOA condensation "dry"
+             Mko(K,JC) = Mk(K,JC)  ! dry mass excl. nh4
+          ENDDO
+       ENDDO
 
+       !debug      if(i==24.and.j==13)       pdbg = .true.
+       CALL TMCOND( OCTAU, XK, Mko, Nko, Mkf, Nkf, SRTOCIL, PDBG, MABS )
+
+       ! ----------- JRP ADD MNFIX...This is for xSOA (JKodros 6/2/15) --------
+       if (pdbg) negvalue=.true. !signal received to printdebug (win, 4/8/06)
+       call mnfix(Nkf,Mkf,negvalue) !<step5.1> bug fix call argument
+       !(win, 4/15/06) !<step4.2> Add call argument to carry tell where mnfix
+       !found
+       if(negvalue) STOP 'MNFIX terminate' !(win, 9/12/05)
+       ! the negative value (win, 9/12/05)
+       !-----------------------------------------------------------------------
+
+       IF( PDBG ) THEN
+          !print 12, I,J,L
+12        FORMAT( 'Error in SOAcond at ', 3I4 )
+          if( .not. SPINUP(60.) )write(116,*) 'Error in SOACOND at',i,j,l
        ELSE
-          OCTAU(K) = 0.e+0_fp
+          PDBG = .false.
```

Comment:
I think the ELSE is here superfluous. PDBG is already FALSE to get to the ELSE block.